### ISSUE TITLE
[2.7.x] [Scala 2.13] routes-compiler artifact not published for 2.7.4 and 2.7.5

### ISSUE THREAD

**Issue by scoquelin** [playframework/playframework#10333]
It seems that only version `2.7.3` has been published for `routes-compiler_2.13`

See [https://search.maven.org/artifact/com.typesafe.play/routes-compiler_2.13](https://search.maven.org/artifact/com.typesafe.play/routes-compiler_2.13)

Ready to help and open a PR if someone can point me in the right direction :wink: 

Thanks!

**Comment by mkurz** [comment]
Indeed, that looks strange. For Scala 2.12 it has been published:
- https://repo1.maven.org/maven2/com/typesafe/play/routes-compiler_2.12/2.7.4/
- https://repo1.maven.org/maven2/com/typesafe/play/routes-compiler_2.12/2.7.5/

For 2.13 it was not:
- https://repo1.maven.org/maven2/com/typesafe/play/routes-compiler_2.13/

@ignasi35 you cut the releases, seems something went wrong here?

**Comment by ignasi35** [comment]
> @ignasi35 you cut the releases, seems something went wrong here?

I'd have to investigate. Maybe a bug in `interplay` causing the `routes-compiler_2.13` artifact to not be produced?

**Comment by ignasi35** [comment]
hi @scoquelin, I think the first step is to `git bisect` to try and isolate the cause. Also, we could simply bump the sbt version in `project/build.properties` to see if that already fixes the issue. In any case, try running `+ publishLocal` to see what's produced.

**Comment by ignasi35** [comment]
> Maybe a bug in `interplay` causing the `routes-compiler_2.13` artifact to not be produced?

I think the problem is in bebf208de

**Comment by ignasi35** [comment]
Wait a second... is this really a bug?
I've just run `sbt new playframework/play-scala-seed.g8` and set the project to use `2.7.5` and scala 2.13.2. It works as expected.

The reason it works is that `routest-compiler` is used from the `sbt-plugin` which runs on scala 2.12 (because it's sbt 1.x). The routes compiler produces scala code which the scala compiler (in this case the version I chose) will produce the bytecode.

I think the error is not that `routes-compiler` is missing in 2.7.4 and 2.7.5 but the fact that it is available in 2.8.x.

Unless `routes-compiler` is being used outside of the `sbt-plugin` which I can't recall at the moment.

Am I missing something?

**Comment by ignasi35** [comment]
@scoquelin if you want to investigate further, a faster method to locate the commit is using `sbt "show Routes-Compiler/crossScalaVersions"`

**Comment by ignasi35** [comment]
Soem more tips: https://github.com/playframework/interplay/compare/2.0.8..2.1.1

**Comment by scoquelin** [comment]
@ignasi35 Some more insights on the use case :

we are using [https://github.com/gradle/playframework](https://github.com/gradle/playframework) to build our play application and it is relying on the `routes-compiler` artifact in order to build its own `play-routes-plugin` [here](https://github.com/gradle/playframework/blob/423796bbc15d278fc8710331e48de1c8287ddb06/build.gradle.kts#L49) 

so it would be great to have a way to republish those artifacts at least for 2.7.x

I will check with your instructions if I can troublehoot this.

**Comment by scoquelin** [comment]
@mkurz @ignasi35 thanks for the hints ! 

so this [change](https://github.com/playframework/interplay/compare/2.0.8..2.1.1#diff-42fd8f4837cde6143515264603c35946R228) in `interplay`  triggered the different behaviour after `2.7.3`

Basically in `2.7.x` branch the output of `sbt "++ 2.13.2 -v"` gives : 

```
[info] Switching Scala version on:
...
[info] Excluding projects:
...
[info]     Routes-Compiler (2.12.11, 2.12.11)
...
```

and after reverting this [change](https://github.com/playframework/interplay/compare/2.0.8..2.1.1#diff-42fd8f4837cde6143515264603c35946R228) in interplay `2.1.x` the output of `sbt "++ 2.13.2 -v"` gives : 

```
[info] Switching Scala version on:
...
[info]     Routes-Compiler (2.12.11, 2.13.2)
...
[info] Excluding projects:
...
```

Now to fix that particular issue I can either : 
- revert the change in `interplay` branch `2.1.x` and we bump interplay version for play `2.7.6`
- simply add `crossScalaVersions` settings [here](https://github.com/playframework/playframework/blob/2.7.x/project/BuildSettings.scala#L258) knowing that `routes-compiler` is the only `PlayDevelopmentProject` anyway (maybe safer?)

Your call let me know and I will open a PR ! Thanks !

**Comment by ignasi35** [comment]
The particular needs of the routes compiler (it's a library for an sbt plugin, but it's used in other plugins so it must be cross compiled) mean there's no recipee in `interplay` for it.

>     * simply add `crossScalaVersions` settings in **2.7.x** [here](https://github.com/playframework/playframework/blob/2.7.x/project/BuildSettings.scala?rgh-link-date=2020-06-20T06%3A12%3A47Z#L258) knowing that `routes-compiler` is the only `PlayDevelopmentProject` anyway (maybe safer?)

I think that's the best move forward. We're in the process of getting rid of `interplay` (see how other projects in the play ecosystem already dropped it). I think overwriting the `crossScalaVersion` and adding a comment with the importance of that overwrite and a link to this discussion should be enough.

**Comment by mkurz** [comment]
@scoquelin Can we closed this issue? I think it's fixed by #10354?
